src/encode/pattern/mod.rs

```diff
@@ -135,17 +135,131 @@ use crate::config::{Deserialize, Deserializers};
 
 mod parser;
 
+const DEFAULT_PATTERN_ENCODER: &str = "{d} {l} {t} - {m}{n}";
+
+#[allow(dead_code)]
+fn default_pattern() -> Option<String> {
+    Some(DEFAULT_PATTERN_ENCODER.to_owned())
+}
+
+/// A simple color map struct
+///
+/// You can use this struct to define a custom color map.
+/// This can be done using a serializer (i.e. config file) or pragmatically.
+///
+/// It is part of the encoder
+///
+/// ```not_rust
+///   pattern: "%m"
+///   color_map:
+///     INFO: Blue
+///     TRACE: Black
+/// ```
+///
+/// Or progmatically in conjuction with PatternEncoder
+///! ```no_run
+///! # #[cfg(all(feature = "console_appender",
+///! #           feature = "file_appender",
+///! #           feature = "pattern_encoder"))]
+///! # fn f() {
+///! let mut log_color_map = ColorMap::default()
+///! log_color_map.info_color = Some(log4rs::encode::Color::Cyan);
+///! }
+///! # }
+///! # fn main() {}
+/// ```
+#[cfg_attr(feature = "config_parsing", derive(serde::Deserialize))]
+#[cfg_attr(feature = "config_parsing", serde(deny_unknown_fields))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ColorMap {
```

Comment:
I guess `Option<Color>` is ok, but that leaves us to only the few defined in the Color struct. 
That would require to add a new entry in the struct Color for arbitrary string, that would be a color such as defined by `colorful` or other module for instance.